iasl: drop package

kidonng opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

A retry of #101821

Replace all instances of `iasl` with `acpica-tools`, and additionally:

- Remove it from `edk2`'s deps since it isn't used
- Revert 59ab29fe88b06b31de5bea7ac11492a9dbc0c680 due to the note:

  > This should be removed again when updating virtualbox to 6.1. 

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

kidonng committed:
iasl: drop pacakge

r-rmcgibbo commented:
Result of `nixpkgs-review pr 133120` at bd5d8cee run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages marked as broken and skipped:</summary>
  <ul>
    <li>OVMF-CSM</li>
    <li>coreboot-utils</li>
    <li>libvmi</li>
    <li>qemu_xen</li>
    <li>qemu_xen-light</li>
    <li>qemu_xen_4_10</li>
    <li>qemu_xen_4_10-light</li>
    <li>qubes-core-vchan-xen</li>
  </ul>
</details>
<details>
  <summary>5 packages built successfully:</summary>
  <ul>
    <li>OVMF</li>
    <li>OVMF-secureBoot</li>
    <li>acpica-tools</li>
    <li>acpidump-all</li>
    <li>fwts</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/tools/misc/coreboot-utils/default.nix:92:7:
```
   |
92 |       installPhase = "install -Dm755 acpidump-all $out/bin/acpidump-all";
   |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/tools/misc/coreboot-utils/default.nix:9:5:
```
  |
9 |     license = licenses.gpl2;
  |     ^
```
</li>
  </ul>
</details>

kidonng committed:
coreboot-utils: add missing phase hooks

Also fix `unclear-gpl`: https://review.coreboot.org/plugins/gitiles/coreboot/+/refs/heads/master#copyright-and-license

SuperSandro2000 commented:
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 133120` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
4 packages marked as broken and skipped:
</summary>
<ul>
<li>linuxPackages-libre.virtualbox</li>
<li>linuxPackages_hardkernel_4_14.virtualbox</li>
<li>linuxPackages_hardkernel_latest.virtualbox</li>
<li>linuxPackages_latest-libre.virtualbox</li>
</ul>
</details>
<details>
<summary>
33 packages built:
</summary>
<ul>
<li>OVMF</li>
<li>OVMF-CSM</li>
<li>OVMF-secureBoot</li>
<li>acpica-tools</li>
<li>acpidump-all</li>
<li>coreboot-utils</li>
<li>fwts</li>
<li>libvmi</li>
<li>linuxPackages.virtualbox (linuxPackages_5_10.virtualbox)</li>
<li>linuxPackages_4_14.virtualbox</li>
<li>linuxPackages_4_19.virtualbox</li>
<li>linuxPackages_4_4.virtualbox</li>
<li>linuxPackages_4_9.virtualbox</li>
<li>linuxPackages_5_12.virtualbox</li>
<li>linuxPackages_5_13.virtualbox (linuxPackages_latest.virtualbox)</li>
<li>linuxPackages_5_4.virtualbox</li>
<li>linuxPackages_hardened.virtualbox</li>
<li>linuxPackages_latest_hardened.virtualbox</li>
<li>linuxPackages_lqx.virtualbox</li>
<li>linuxPackages_testing_bcachefs.virtualbox</li>
<li>linuxPackages_xanmod.virtualbox</li>
<li>linuxPackages_zen.virtualbox</li>
<li>qemu_xen (qemu_xen_4_10)</li>
<li>qemu_xen-light (qemu_xen_4_10-light)</li>
<li>qubes-core-vchan-xen</li>
<li>seabios</li>
<li>virtualbox</li>
<li>virtualboxHardened</li>
<li>virtualboxHeadless</li>
<li>virtualboxWithExtpack</li>
<li>xen (xenPackages.xen-vanilla ,xenPackages.xen_4_10-vanilla ,xen_4_10)</li>
<li>xen-light (xenPackages.xen-light ,xenPackages.xen_4_10-light ,xen_4_10-light)</li>
<li>xen-slim (xenPackages.xen-slim ,xenPackages.xen_4_10-slim ,xen_4_10-slim)</li>
</ul>
</details>
